### PR TITLE
Feature/user model

### PR DESCRIPTION
**Pull request by idyassineSmail**
I add some APIs:
GET: List of all users
GET: one user by id
PUT: Update user
DELETE: Delete user by id

### CODE REVIEW

File: src/user/user.service.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  id: number;
+  username: string;
+  email: string;
+  bio: string | null;
+  profileImage: string | null;
+  coverImage: string | null;
+};
+
+// Find all users
+export const listUsers = async (): Promise<User[]> => {
```
Comment (ZakariaBaqasse) [inline src/user/user.service.ts:15]: The call to the Prisma ORM methods must be done in a user.model.ts file and the service must only make call to those methods of the model and handle any potential errors/ exceptions that might arise.

File: src/user/user.router.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import type { Request, Response } from "express";
+import { body, validationResult } from "express-validator";
+
+import * as UserService from "./user.service";
+
+export const userRouter = express.Router();
+
+//GET: List of all users
+userRouter.get("/", async (request: Request, response: Response) => {
+  try {
```
Comment (ZakariaBaqasse) [inline src/user/user.router.ts:11]: this try catch logic must be moved to the service. The router/controller mustn't contain any logic it just map the request to the appropriate service functions

**Review by ZakariaBaqasse — CHANGES_REQUESTED** [review]

File: src/user/user.router.ts (hunk 2 above)
Comment (idyassineSmail) [reply src/user/user.router.ts]: Done!

File: src/user/user.service.ts (hunk 1 above)
Comment (idyassineSmail) [reply src/user/user.service.ts]: Done!

**Review by ZakariaBaqasse — APPROVED** [review]
all good now

**Pull request merged**